--- /root/tmp/kuryr-kubernetes/kuryr_kubernetes/k8s_client.py	2017-06-19 07:15:39.901398831 -0400
+++ kuryr_kubernetes/k8s_client.py	2017-06-22 06:14:48.177325667 -0400
@@ -138,7 +138,7 @@
                                  headers=header)) as response:
                 if not response.ok:
                     raise exc.K8sClientException(response.text)
-                for line in response.iter_lines(delimiter='\n'):
+                for line in response.iter_lines(delimiter=b'\n'):
                     line = line.strip()
                     if line:
                         yield jsonutils.loads(line)
